src/Margay.cpp

```diff
@@ -715,22 +729,25 @@ void Margay::blinkGood()
 
 float Margay::getVoltage()  //Get voltage from Ax pin
 {
-	// first allow access to ADC via internal I2C
-	pinMode(I2C_SW, INPUT);
-	bool extI2COn = digitalRead(SD_CD);
-
-	if ( !extI2COn ){
-		initADC(); // initialize ADC
-		float val = adc.GetVoltage();
-		return val;
-	}
-	else{
-		externalI2C(OFF);
-		initADC(); // initialize ADC
-		float val = adc.GetVoltage();
-		externalI2C(ON); // re-allow external I2C		
-		return val;
-	}
+	// voltage reads from the on-board ADC, but to read the Ax pin at the same time as external
+	// sensors, need access to the ADC. However, we do not want to change the state of the I2C 
+	// bus communication by taking a voltage reading. So we have logic here to make the switch.
+
+	// first check whether external I2C connnections are on by testing pin I2C_SW (HIGH is on)
+	bool initialStateI2C = digitalRead(I2C_SW);
+
+	// initialize a variable to hold the voltage reading.
+	float val = 0;
+
+	switchExternalI2C(OFF);
+	initADC(18); // initialize ADC
+	val = adc.GetVoltage();
+	switchExternalI2C(ON); // re-allow external I2C		
+
+	// make sure I2C Bus is returned to initial state
+	closeFarmGateI2C(initialStateI2C);
```

Comment:
Can open or close gate, so:
```suggestion
	farmGateI2C(initialStateExternI2C);
```